lib/matchers/associations.rb

```diff
@@ -38,8 +37,8 @@ def ordered_by(association_field_name)
           raise "#{@association[:type].inspect} does not respond to :order" unless [HAS_MANY, HAS_AND_BELONGS_TO_MANY, EMBEDS_MANY].include?(@association[:type])
           @association[:order] = association_field_name.to_s
           @expectation_message << " ordered by #{@association[:order].inspect}"
-
-          if association_field_name.is_a? Origin::Key
+          query_key = (Mongoid::VERSION =~ /^6/).blank? ? Origin::Key : Mongoid::Criteria::Queryable::Key
+          if association_field_name.is_a? query_key
```

Comment:
Thanks for you contribution. 👏 

Could we just improve it a little to make it more readable? Maybe move the condition to a specific a predicate method and call it on another method that returns the correct `key`. What do you think about it?